keras_nlp/src/models/task.py

```diff
@@ -146,6 +146,7 @@ def from_preset(
         cls,
         preset,
         load_weights=True,
+        load_task_extras=False,
```

Comment:
My biggest reason for wanting to flip the default is to have parallel quickstarts for vision and text models:

```python
classifier = TextClassifier.from_preset(
    "bert_base_en",
    num_classes=2,
)
classifier.fit(text_dataset)

classifier = ImageClassifier.from_preset(
    "res_net_50",
    num_classes=2,
)
classifier.fit(image_dataset)
```

I think those snippets are important, they will be front and center. We should avoid needing to introduce more concepts there.

This also flips the arg to be more explicit (explicit is good I think). Passing true will now error if a `task.json` does not exist, or if it is for the wrong class. But we definitely cannot keep this strict behavior if we flip the default. It would break our [current quickstart](https://github.com/keras-team/keras-nlp?tab=readme-ov-file#quickstart)!